server/src/main/java/org/elasticsearch/inference/ModelConfigurations.java

```diff
@@ -61,7 +73,12 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeEnum(taskType);
         out.writeString(service);
         out.writeNamedWriteable(serviceSettings);
-        out.writeNamedWriteable(taskSettings);
+
+        if (out.getTransportVersion().onOrAfter(ML_INFERENCE_TASK_SETTINGS_OPTIONAL_ADDED)) {
+            out.writeOptionalNamedWriteable(taskSettings);
+        } else {
+            out.writeNamedWriteable(taskSettings);
```

Comment:
Ah good point.

What I was trying to accomplish was avoiding having to create a unique class that doesn't do anything for each service that won't have any task settings.

Actually maybe what I'll do is create a class that represents an empty task settings, then if `ModelConfigurations` receives null, it'll write that class instead. I think that way we can avoid having to change the transport version.

Reply:
I think you'll have to change the transport version whatever you do, because 8.11 and earlier only know about `ElserMlNodeTaskSettings`. They won't know about any new class you add now.

Reply:
I think the best thing is that `InferenceService` has `getMinimalSupportedTransportVersion` method, and then we check that against `ClusterState.getMinTransportVersion()` when first putting a model around here: https://github.com/elastic/elasticsearch/blob/0848c2b85bb9ffeecfa9dff4f6d9a2e9b13b0b0d/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportPutInferenceModelAction.java#L94

Doing that should mean that older nodes never need to handle model types they don't understand either via transport, or, just as importantly, by loading persisted configs.

It's easiest all round if we don't let people use brand new model types until the whole cluster understands them.